src/logic/graphMapping.ts

```diff
@@ -0,0 +1,115 @@
+
+import { Graph, MappedEdge, MappedNode } from './commonInterfaces';
+
+const LABEL_ALPHABET = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+
+export class Edge {
+  label?: number;
+  from: number;
+  to: number;
+  equals = (other: Edge) => {
+    return this.label === other.label && (
+      (this.from === other.from && this.to === other.to) ||
+      (this.from === other.to && this.to === other.from)
+    )
+  };
+  constructor(from: number, to: number, label?: number) {
+    this.from = from;
+    this.to = to;
+    this.label = label;
+  }
+};
+
+/**
+ * Finds all edges in graph matrix. This maybe inefficient (complexity of
+ * O(n^3)), but for this purpose it's ok. Tested up to 2016 edges. For
+ * that amount takes < 1s. Maybe a bottleneck in the future.
+ */
+export function getEdgesArray(graph: Graph): Array<Edge> {
+  const edges: Set<Edge> = new Set();
+  const isAccounted = (edge: Edge) => {
+    for (const e of edges) {
+      if (e.equals(edge)) return true;
```

Comment:
I'm pretty sure you can take advantage of `Array.some` here

Reply:
I probably could, but this function is a mess. I'm not sure if casting to an array is a good idea here, because it probably iterates over whole set, and `some` probably does to. This part is critical in terms of performance. Even though build-in functions are much faster this iterates only once, so performance gain may be minimal or none at all.
In order to properly use `Array.some` I would need to keep edges in array in the first place. This is good idea, since I used `Set` thinking it would remove duplicates, but I forgot it doesn't work because edges are references and `Set` only removes duplicate pointers.

Reply:
Idk, this function needs more love than just this. Cutting off half of graph would solve duplicates problem. So I will just leave it like that maybe in future it will be rewritten in more elegant way.

Reply:
Ow, I missed it's a `Set` tbh. Ideally, we should use `Set.has` here but if u think it requires an extra job then let's proceed with this.